grpc/percent_slot_test.go

```diff
@@ -0,0 +1,130 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package grpc_test
+
+import (
+	. "github.com/DataDog/chaos-controller/grpc"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("get mapping from randomly generated Percent to Alteration based on AlterationToPercentAffected", func() {
+	var (
+		altCfgs             []AlterationConfiguration
+		mapping             map[AlterationConfiguration]PercentAffected
+		percentToAlteration []AlterationConfiguration
+	)
+	BeforeEach(func() {
+		altCfgs = []AlterationConfiguration{
+			{
+				ErrorToReturn:    "CANCELED",
+				OverrideToReturn: "",
+			},
+			{
+				ErrorToReturn:    "ALREADY_EXISTS",
+				OverrideToReturn: "",
+			},
+			{
+				ErrorToReturn:    "",
+				OverrideToReturn: "{}",
+			},
+		}
+		mapping = make(map[AlterationConfiguration]PercentAffected)
+	})
+
+	Context("with one alteration", func() {
+		BeforeEach(func() {
+			mapping[altCfgs[0]] = PercentAffected(15)
+
+			percentToAlteration = GetPercentToAlteration(mapping)
+		})
+
+		It("should create 15 SlotPercents", func() {
+			Expect(len(percentToAlteration)).To(Equal(15))
+		})
+
+		It("should create 15 Slot Percents which are configured to the single alteration", func() {
+			altCfg_0 := percentToAlteration[0]
+			Expect(altCfg_0.ErrorToReturn).To(Equal("CANCELED"))
+			Expect(altCfg_0.OverrideToReturn).To(Equal(""))
+
+			altCfg_14 := percentToAlteration[14]
+			Expect(altCfg_14.ErrorToReturn).To(Equal("CANCELED"))
+			Expect(altCfg_14.OverrideToReturn).To(Equal(""))
+		})
+	})
+
+	Context("with three alterations", func() {
+		BeforeEach(func() {
+			mapping[altCfgs[0]] = PercentAffected(15)
+			mapping[altCfgs[1]] = PercentAffected(20)
+			mapping[altCfgs[2]] = PercentAffected(35)
+
+			percentToAlteration = GetPercentToAlteration(mapping)
+		})
+
+		It("should create 70 SlotPercents", func() {
```

Comment:
what is a slotpercent, here?

Reply:
tried to make it make more sense with the x out of 100